admin/resources/js/components/pool/UpdatePool.tsx

```diff
@@ -0,0 +1,271 @@
+import {
+  Classification,
+  CmoAsset,
+  OperationalRequirement,
+  Pool,
+  UpdatePoolInput,
+  UpdatePoolMutation,
+  useGetUpdatePoolDataQuery,
+  User,
+  useUpdatePoolMutation,
+} from "../../api/generated";
+import { notEmpty } from "../../helpers/util";
+import errorMessages from "../form/errorMessages";
+import { unpackIds } from "../form/formUtils";
+import Input from "../form/Input";
+import MultiSelect from "../form/MultiSelect";
+import Select from "../form/Select";
+import Submit from "../form/Submit";
+import TextArea from "../form/TextArea";
+
+type Option<V> = { value: V; label: string };
+
+type FormValues = Pick<Pool, "name" | "description"> & {
+  assetCriteria: string[] | undefined;
+  classifications: string[] | undefined;
+  essentialCriteria: string[] | undefined;
+  operationalRequirements: string[] | undefined;
+  owner: string;
+};
+
+interface UpdatePoolFormProps {
+  classifications: Classification[];
+  cmoAssets: CmoAsset[];
+  initialPool: Pool;
+  locale: "en" | "fr";
+  operationalRequirements: OperationalRequirement[];
+  users: User[];
+  handleUpdatePool: (
+    id: string,
+    data: UpdatePoolInput,
+  ) => Promise<UpdatePoolMutation["updatePool"]>;
+}
+
+export const UpdatePoolForm: React.FunctionComponent<UpdatePoolFormProps> = ({
+  classifications,
+  cmoAssets,
+  initialPool,
+  locale,
+  operationalRequirements,
+  users,
+  handleUpdatePool,
+}) => {
+  const intl = useIntl();
+  const dataToFormValues = (
+    data: Pool | UpdatePoolMutation["updatePool"],
+  ): FormValues => ({
+    ...data,
+    assetCriteria: unpackIds(data?.assetCriteria),
+    classifications: unpackIds(data?.classifications),
+    essentialCriteria: unpackIds(data?.essentialCriteria),
+    operationalRequirements: unpackIds(data?.operationalRequirements),
+    owner: data?.owner?.id || "",
+  });
+
+  const formValuesToSubmitData = (values: FormValues): UpdatePoolInput => ({
+    ...values,
+    assetCriteria: {
+      sync: values.assetCriteria,
+    },
+    classifications: {
+      sync: values.classifications,
+    },
+    essentialCriteria: {
+      sync: values.essentialCriteria,
+    },
+    operationalRequirements: {
+      sync: values.operationalRequirements,
+    },
+    owner: { connect: values.owner },
+    name: {
+      en: values.name?.en,
+      fr: values.name?.fr,
+    },
+    description: {
+      en: values.description?.en,
+      fr: values.description?.fr,
+    },
+  });
+
+  const methods = useForm<FormValues>({
+    defaultValues: dataToFormValues(initialPool),
+  });
+  const { handleSubmit } = methods;
+
+  const onSubmit: SubmitHandler<FormValues> = async (data: FormValues) => {
+    console.log(formValuesToSubmitData(data));
+    await handleUpdatePool(initialPool.id, formValuesToSubmitData(data))
+      .then(() => {
+        // TODO: Navigate to pool candidate dashboard.
+      })
+      .catch(() => {
+        // Something went wrong with handleCreatePoolCandidate.
```

Comment:
The comment should reference handleCreatePool instead of handleCreatePoolCandidate.